### ISSUE TITLE
Pipeline task runs but its `dependsOn` tasks are ignored

### ISSUE THREAD

**Issue by oedotme** [vercel/turborepo#135]
### What version of Turbrepo are you using?

0.8.5 - 1.0.1

### Describe the Bug

Pipeline task runs while its `dependsOn` tasks stopped working from version `0.8.5` and higher.

Working as expected in `0.8.4`

### Expected Behavior

The defined `dependsOn` tasks list should run before the main pipeline task runs.

#### Environment 
- Linux 
- Node `17.1.0`
- NPM `8.1.3`

#### Config
- NPM workspaces
```json
  "workspaces": [
    "apps/*",
    "libs/*"
  ],
```

- Turbo
```json
  "turbo": {
    "bashBranch": "origin/main",
    "pipeline": {
      "dev": {
        "cache": false
      },
      "lint": {},
      "test": {
        "dependsOn": [
          "lint"
        ]
      },
      "build": {
        "dependsOn": [
          "lint",
          "test"
        ],
        "outputs": [
          "build/**",
          "dist/**",
          ".next/**",
          "!.next/cache/**"
        ]
      }
    }
  },
```

**Comment by jaredpalmer** [comment]
Fixed in 1.0.6

**Comment by charles-salmon** [comment]
@jaredpalmer I was just trying Turborepo out, and encountered this same issue in v1.0.6.

I was using the following configuration:

```
{
  "turbo": {
    "baseBranch": "origin/main",
    "pipeline": {
      "format": {},
      "generate": {
        "outputs": [
          ".gen/**"
        ]
      },
      "build": {
        "dependsOn": [
          "^build",
          "generate"
        ]
      }
    }
  }
}
```

and observed that a task named `infrastructure:build` was started immediately after `infrastructure:generate` was started (prior to code generation having been completed).

In case it provides any additional context for you to work from, in the `infrastructure:generate` task, I was using the `cdktf` CLI to generate TS definitions from a Terraform module, using the `cdktf get` command.
